src/Stripe/Infrastructure/Requestor.cs

```diff
@@ -1,106 +1,182 @@
-﻿using System;
-using System.IO;
-using System.Net;
-using System.Text;
-
-namespace Stripe
-{
-    internal static class Requestor
-    {
-        public static string GetString(string url, string apiKey = null)
-        {
-            var wr = GetWebRequest(url, "GET", apiKey);
-
-            return ExecuteWebRequest(wr);
-        }
-
-        public static string PostString(string url, string apiKey = null)
-        {
-            var wr = GetWebRequest(url, "POST", apiKey);
-
-            return ExecuteWebRequest(wr);
-        }
-
-        public static string PostStringBearer(string url, string apiKey = null)
-        {
-            var wr = GetWebRequest(url, "POST", apiKey, true);
-
-            return ExecuteWebRequest(wr);
-        }
-
-        public static string Delete(string url, string apiKey = null)
-        {
-            var wr = GetWebRequest(url, "DELETE", apiKey);
-
-            return ExecuteWebRequest(wr);
-        }
-
-        internal static WebRequest GetWebRequest(string url, string method, string apiKey = null, bool useBearer = false)
-        {
-            apiKey = apiKey ?? StripeConfiguration.GetApiKey();
-
-            var request = (HttpWebRequest)WebRequest.Create(url);
-            request.Method = method;
-
-            if(!useBearer)
-                request.Headers.Add("Authorization", GetAuthorizationHeaderValue(apiKey));
-            else
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Text;
+
+#if NET40
+using System.Threading.Tasks;
+#endif
+
+namespace Stripe
+{
+    internal static class Requestor
+    {
+        public static string GetString(string url, string apiKey = null)
+        {
+            var wr = GetWebRequest(url, "GET", apiKey);

```

Comment:
@jt000 Not only does that not work with 3.5, but it is best not to mix async and blocking code, in particular calling Result or GetResult() from synchronous code without paying attention to the captured context can result in a deadlock. More info here: https://msdn.microsoft.com/en-us/magazine/jj991977.aspx